parquet/src/test/java/org/apache/iceberg/avro/TestReadProjection.java

```diff
@@ -513,17 +513,26 @@ public void testListOfStructsProjection() throws IOException {
     );
 
     projected = writeAndRead("y_renamed", writeSchema, yRenamed, record);
-    Assert.assertNull("Should not project id", projected.get("id"));
+    assertNotProjected("Should not project id", projected, "id");
     Assert.assertNotNull("Should project points list", projected.get("points"));
     points = (List<Record>) projected.get("points");
     Assert.assertEquals("Should read 2 points", 2, points.size());
     projectedP1 = points.get(0);
-    Assert.assertNull("Should not project x", projectedP1.get("x"));
-    Assert.assertNull("Should not project y", projectedP1.get("y"));
+    assertNotProjected("Should not project x", projectedP1, "x");
+    assertNotProjected("Should not project y", projectedP1, "y");
     Assert.assertEquals("Should project z", 2, (int) projectedP1.get("z"));
     projectedP2 = points.get(1);
-    Assert.assertNull("Should not project x", projectedP2.get("x"));
-    Assert.assertNull("Should not project y", projectedP2.get("y"));
+    assertNotProjected("Should not project x", projectedP2, "x");
+    assertNotProjected("Should not project y", projectedP2, "y");
     Assert.assertEquals("Should project null z", null, projectedP2.get("z"));
   }
+
+  public static void assertNotProjected(String message, Record projected, String fieldName) {
+    try {
+      projected.get(fieldName);
+      Assert.fail(message);
```

Comment:
I feel we don't need this line.

Reply:
this is to mimic the `Assert.assertNull` w/ error message that this API replaces.